rfcs/0008-leaderless-dns.md

```diff
@@ -0,0 +1,238 @@
+
+If all local and remote health checks for a particular host are marked as unhealthy, all known IPs are published, to avoid an NXDOMAIN response.
+
+## OCM / ACM
+This is very similar to a single cluster, with the caveat that the "single" cluster is the hub cluster for this OCM. The primary difference is that the gateway will be a Kuadrant gatewayClass.
+
+### OCM / ACM Health Checks
+This is also the same as single cluster.
+
+## Migrating from Single cluster to Multiple cluster without OCM
+
+In the single cluster: 
+- Edit all relevant provider secrets and set:
+  - `multiClusterID` to some unique value
+
+In the additional clusters:
+- Copy the DNS Policies and provider secrets from the single cluster to any additional clusters, and update the provider secrets with unique `multiClusterID` values.
+- (optional) Create a gateway and workload which uses the same hosts as the single cluster.
+
+## Migrating from single/multiple clusters to OCM
+
+This is not covered.
+
+## Cleanup
+
+Cleanup of a cluster's zone file entries is triggered by deleting the DNS Policy that caused the DNS Records to be created.
+
+The DNS policy will have a finalizer, when it is deleted the controller will mark the created DNS Records for deletion.
+
+The DNS Records will have a finalizer from the DNS Operator which will clean the records from the zone before removing the finalizer.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## General Logic Overview
+
+The general flow for in the Kuadrant operator follows a single path, where it will always output a DNS Record which specifies everything needed for this workload on this cluster, and is unaware of the concept of distributed DNS, or phrased more simply:
+Build a DNS Record (let's refer to this as a kuadrantRecord) based on the local gateways and DNS Policy.
+
+This kuadrantRecord is reconciled by the DNS Operator, where there are 2 possible routes:
+
+Where a multiClusterID is specified in the relevant ProviderSecret the DNS Operator will act with the DNS Provider's zone as the authority and ensure that the records in the zone relevant to the hosts defined within gateways on it's cluster are accurate. When cleaning up a DNS Record in this scenario the operator will ensure all records owned by the policy for the gateway on the local cluster are removed and then prune unresolvable records related to the hosts defined in the gateway from the DNS Provider zone.
+
+Where a multiClusterID is not specified the DNS Operator will act with the DNS Record as the authority and ensure the entire host is updated in the zone to exactly match the specified DNS Record when cleaning up in this scenario, the host is fully removed from the DNS Provider zone.
+
+### API Updates
+New configuration options in the providerSecret:
+- `multiClusterID`
+  - When absent the DNS Operator will consider itself the owner of the zone and the local DNS Records as the authority.
```

Comment:
What if only one cluster doesn't have `multiclusterID`, will he break it for the others?

Reply:
yes, as Craig mentioned above, this is bad configuration, and the cluster without the absent will act as the authority and remove all the other clusters records, until it is correctly configured.

Reply:
we will not be exposing this value to users now. Resolving this comment.

Reply:
I would like to reopen this because I believe this will be a big usability problem and makes it very easy for one small issue to mess everything up. And whats worse it would be impossible to report this broken status back, meaning users won't know it is broken until they try it out.

Reply:
This field is going away, if see this comment tree: https://github.com/Kuadrant/architecture/pull/55#discussion_r1487460320

This is why I have closed this comment, I am rewriting the RFC to reflect this change.

I'm going to reclose this, but if you still have concerns after reading the above comments, then let's discuss it there to keep everything together.